examples/rancher-gen/nginx.tmpl

```diff
@@ -0,0 +1,183 @@
+{{$local := host}}
+#This Host:
+#  UUID: {{$local.UUID}}
+#  Name: {{$local.Name}}
+#  Address: {{$local.Address}}
+#  Hostname: {{$local.Hostname}}
+#  Labels: {{$local.Labels}}
+
+# Access Log Formatting
+log_format vhost '$host $remote_addr - $remote_user [$time_local] '
+                 '"$request" $status $body_bytes_sent '
+                 '"$http_referer" "$http_user_agent"';
+
+
+# If we receive X-Forwarded-Proto, pass it through; otherwise, pass along the
+# scheme used to connect to this server
+map $http_x_forwarded_proto $proxy_x_forwarded_proto {
+  default $http_x_forwarded_proto;
+  ''      $scheme;
+}
+
+# If we receive X-Forwarded-Port, pass it through; otherwise, pass along the
+# server port the client connected to
+map $http_x_forwarded_port $proxy_x_forwarded_port {
+  default $http_x_forwarded_port;
+  ''      $server_port;
+}
+
+# If we receive Upgrade, set Connection to "upgrade"; otherwise, delete any
+# Connection header that may have been passed to this server
+map $http_upgrade $proxy_connection {
+  default upgrade;
+  '' close;
+}
+
+gzip_types text/plain text/css application/javascript application/json application/x-javascript text/xml application/xml application/xml+rss text/javascript;
+
+access_log off;
+
+{{ if (exists "/etc/nginx/proxy.conf") }}
+include /etc/nginx/proxy.conf;
+{{ else }}
+# HTTP 1.1 support
+proxy_http_version 1.1;
+proxy_buffering off;
+proxy_set_header Host $http_host;
+proxy_set_header Upgrade $http_upgrade;
+proxy_set_header Connection $proxy_connection;
+proxy_set_header X-Real-IP $remote_addr;
+proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+proxy_set_header X-Forwarded-Proto $proxy_x_forwarded_proto;
+proxy_set_header X-Forwarded-Port $proxy_x_forwarded_port;
+
+# Mitigate httpoxy attack (see README for details)
+proxy_set_header Proxy "";
+{{ end }}
+
+server {
+  server_name _; # This is just an invalid value which will never trigger on a real hostname.
+  listen 80;
+  access_log /var/log/nginx/access.log vhost;
+  return 503;
+}
+
+## ACME CHALLENGE -------
+{{range $labelValue, $services := services | groupByLabel "rgon-acme" -}}
+upstream acmetool {
+  {{range $serv := $services}}{{range $cnt := .Containers -}}
+
+    {{range $port := $serv.Ports}}
+      server {{$cnt.Address}}:{{$port.InternalPort}};
+    {{- end}}
+
+  {{- end}}{{- end}}
+}
+{{end}}
+
+## HOSTS ----------------
+{{range $labelValue, $hosts := hosts | groupByLabel "rgon.domain" -}}
```

Comment:
I use this to proxy Rancher itself! Adding labels to the main Rancher host allows us to access Rancher at https://rancher-test.com instead of http://rancher-test.com:8080.